src/stages/stage5/timeline.c

```diff
@@ -334,197 +135,713 @@ static int stage5_magnetto(Enemy *e, int t) {
+TASK(lightburst_fairy_1, {
+	cmplx pos;
+	MoveParams move_enter;
+	MoveParams move_exit;
+}) {
+	Enemy *e = TASK_BIND(espawn_big_fairy(ARGS.pos, ITEMS(.points = 4, .power = 2)));
+
+	e->move = ARGS.move_enter;
+	INVOKE_SUBTASK_DELAYED(200, lightburst_fairy_move, {
+			.e = ENT_BOX(e),
+			.move = ARGS.move_exit
+	});
+
+	real count = difficulty_value(6, 7, 8, 9);
+	WAIT(70);
+	for (int x = 0; x < 150; x++) {
+		play_sfx("shot1_loop");
+		for (int i = 0; i < count; i++) {
+			cmplx n = cdir(carg(global.plr.pos) + 2.0 * M_PI / count * i);
+			PROJECTILE(
+				.proto = pp_ball,
+				.pos = e->pos + 50 * n * cdir(-0.4 * x * global.diff),
+				.color = RGB(0.3, 0, 0.7),
+				.move = move_asymptotic_simple(3 * n, 3),
+			);
+			play_sfx("shot1_loop");
+		}
+		play_sfx("shot2");
+		WAIT(5);
 	}
+}
 
-	FROM_TO(270, 320, 40)
-		create_enemy1c(VIEWPORT_W/4+VIEWPORT_W/2*_i, 2000, BigFairy, stage5_lightburst, 2.0*I);
+TASK(laser_fairy, {
+	cmplx pos;
+	cmplx acceleration;
+	int reduction;
+}) {
+	Enemy *e = TASK_BIND(espawn_big_fairy(ARGS.pos, ITEMS(.points = 4, .power = 2)));
 
-	FROM_TO(400, 600, 10) {
-		tsrand_fill(2);
-		create_enemy3c(200.0*I*afrand(0), 500, Swirl, stage5_swirl, 4+I, 70+20*afrand(1)+200.0*I, cexp(-0.05*I));
-	}
+	e->move = move_towards(e->pos + ARGS.acceleration * 100, 0.05);
+	WAIT(100);
 
-	FROM_TO(700, 800, 10) {
-		tsrand_fill(3);
-		create_enemy3c(VIEWPORT_W+200.0*I*afrand(0), 500, Swirl, stage5_swirl, -4+afrand(1)*I, 70+20*afrand(2)+200.0*I, cexp(0.05*I));
-	}
+	int delay = difficulty_value(6, 5, 4, 3) + ARGS.reduction;
+	int amount = 700 / delay;
 
-	FROM_TO(870+50*(global.diff==D_Easy), 1000, 50)
-		create_enemy1c(VIEWPORT_W/4+VIEWPORT_W/2*(_i&1), 2000, BigFairy, stage5_limiter, I);
+	for (int x = 0; x < amount; x++) {
+		cmplx n = cdir(carg(global.plr.pos - e->pos) + (0.2 - 0.02 * global.diff) * x);
+		float fac = (0.5 + 0.2 * global.diff);
 
-	AT(1000)
-		create_enemy1c(VIEWPORT_W/2, 9000, BigFairy, stage5_laserfairy, 2.0*I);
+		// TODO: is this the correct "modern" way of invoking lasers?
+		create_lasercurve2c(e->pos, 100, 300, RGBA(0.7, 0.3, 1, 0), las_accel, fac * 4 * n, fac * 0.05 * n);
 
-	FROM_TO(1400+100*(D_Lunatic - global.diff), 2560, 60-5*global.diff) {
-		tsrand_fill(2);
-		create_enemy1c(VIEWPORT_W+200.0*I*afrand(0), 500, Swirl, stage5_miner, -3+2.0*I*afrand(1));
+		PROJECTILE(
+			.proto = pp_plainball,
+			.pos = e->pos,
+			.color = RGBA(0.7, 0.3, 1, 0),
+			.move = move_accelerated(fac * 4 * n, fac * 0.05 * n),
+		);
+		play_sfx_ex("shot_special1", 0, true);
+		WAIT(delay);
 	}
 
-	FROM_TO(1500, 2400, 80)
-		create_enemy1c(VIEWPORT_W*(_i&1)+100.0*I, 300, Fairy, stage5_greeter, 3-6*(_i&1));
+	WAIT(30);
+	e->move = move_linear(-ARGS.acceleration);
+}
 
-	AT(2500) {
-		create_enemy1c(VIEWPORT_W/2, 2000, BigFairy, stage5_lightburst, 2.0*I);
+TASK(lightburst_fairies_1, {
+	int num;
+	cmplx pos1;
+	cmplx pos2;
+	cmplx exit;
+}) {
+	for (int i = 0; i < ARGS.num; i++) {
+		cmplx pos = ARGS.pos1 + ARGS.pos2 * i;
```

Comment:
Those parameter names seem misleading. You'd think `pos1` and `pos2` are both absolute positions, but in fact `pos2` seems to be used like an offset to `pos1`.